tests/openvino/test_modeling.py

```diff
@@ -727,7 +728,8 @@ def test_compare_to_transformers(self, model_arch):
         self.assertTrue(torch.allclose(ov_outputs.logits, transformers_outputs.logits, equal_nan=True, atol=1e-4))
 
         # Qwen tokenizer does not support padding
-        if model_arch == "qwen":
+        # Gemma2 result different due to difference in cache representation
```

Comment:
it changes several tokens in generate results, but testing on real model, result looks reasonable

Reply:
> in tests, we use batching where sequences has different length, but in comparison with position_ids, cache_positions does allow to specify separately positions for every element in batch

Could be an issue if this results in mismatch for batched inputs of different length, do you know if we have this issue in the case where use_cache=False ?

Reply:
issue visible only for model with cacne and only if batch > 1 with non-equal sequence length

Reply:
OK thanks for the update, we need to fix this before we can merge, do you know if this could be related to https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/gemma2/modeling_gemma2.py#L891 (we had this issue for batched inputs for llama models at some point)

Reply:
I do not think that it is related. problem as I already repeat several times that original model uses static cache and not good with dynamic cache support. It is required full model rewriting to fix this accuracy issue

Reply:
ok, I added torch model patching to enable dynamic cache in tests to match openvino exported model behaviour